src/Bridge/Symfony/Routing/ApiLoader.php

```diff
@@ -71,21 +83,117 @@ public function load($data, $type = null)
             }
+
+            $this->computeSubresourceOperations($routeCollection, $resourceClass);
         }
 
         return $routeCollection;
     }
 
     /**
+     * Transforms a given string to a tableized, pluralized string.
+     *
+     * @param string $name usually a ResourceMetadata shortname
+     *
+     * @return string A string that is a part of the route name
+     */
+    private function routeNameResolver(string $name, bool $pluralize = true): string
+    {
+        $name = Inflector::tableize($name);
+
+        return $pluralize ? Inflector::pluralize($name) : $name;
+    }
+
+    /**
+     * Handles subresource operations recursively and declare their corresponding routes.
+     *
+     * @param RouteCollection $routeCollection
+     * @param string          $resourceClass
+     * @param string          $rootResourceClass null on the first iteration, it then keeps track of the origin resource class
+     * @param string          $rootResourceClass null on the first iteration, it then keeps track of the parent operation
+     */
+    private function computeSubresourceOperations(RouteCollection $routeCollection, string $resourceClass, $rootResourceClass = null, $parentOperation = null)
+    {
+        if (null === $this->propertyNameCollectionFactory || null === $this->propertyMetadataFactory) {
+            return;
+        }
+
+        if (null === $rootResourceClass) {
+            $rootResourceClass = $resourceClass;
+        }
+
+        foreach ($this->propertyNameCollectionFactory->create($resourceClass) as $property) {
+            $propertyMetadata = $this->propertyMetadataFactory->create($resourceClass, $property);
+
+            if (!$propertyMetadata->hasSubresource()) {
+                continue;
+            }
+
+            $isCollection = $propertyMetadata->getType()->isCollection();
+            $subresource = $isCollection ? $propertyMetadata->getType()->getCollectionValueType()->getClassName() : $propertyMetadata->getType()->getClassName();
+
+            $propertyName = $this->routeNameResolver($property, $isCollection);
+
+            $operation = [
+                'property' => $property,
+                'collection' => $isCollection,
+            ];
+
+            if (null === $parentOperation) {
+                $rootResourceMetadata = $this->resourceMetadataFactory->create($rootResourceClass);
+                $rootShortname = $rootResourceMetadata->getShortName();
+                $resourceRouteName = $this->routeNameResolver($rootShortname);
+
+                $operation['identifiers'] = [['id', $rootResourceClass]];
+                $operation['route_name'] = sprintf('%s%s_%s%s', self::ROUTE_NAME_PREFIX, $resourceRouteName, $propertyName, self::SUBRESOURCE_SUFFIX);
+
+                $operation['path'] = $this->operationPathResolver->resolveOperationPath($rootShortname, $operation, OperationType::SUBRESOURCE);
+            } else {
+                $identifier = $parentOperation['property'];
```

Comment:
seems like it is used just one  time, maybe you shouldn't be creating a variable ?